### PR TITLE
Track unregistered stale connection count

### PR DESCRIPTION
**Pull request by SteveHNH**
RHCLOUD-9224

Signed-off-by: Stephen Adams <tsadams@gmail.com>

### CODE REVIEW

File: internal/controller/redis_connector.go [hunk 1]
```diff
@@ -70,6 +70,7 @@ func UnregisterWithRedis(client *redis.Client, account, nodeID, hostname string)
 	_, err := client.TxPipelined(func(pipe redis.Pipeliner) error {
 		client.Del(getConnectionKey(account, nodeID))
 		removeIndexes(client, account, nodeID, hostname)
+		metrics.unregisterStaleConnectionFromRedis.Inc()
```
Comment (dehort) [inline internal/controller/redis_connector.go:73]: This method is used in the case of normal un-registration so I don't think this is the right place for this counter.

**Review by dehort — CHANGES_REQUESTED** [review]
For RHCLOUD-9224, I was thinking of modifying the connection cleaner util so that it recorded the stale connection cleanup metric.  I added a bit more details to RHCLOUD-9224.  Let me know if we need to discuss.

**Comment by SteveHNH** [comment]
@dehort Give it a look now. I moved the metrics collection bit into connection_cleaner (like I was supposed to), and added prometheus push gateway config.

File: cmd/connection_cleaner/main.go [hunk 2]
```diff
@@ -33,18 +40,36 @@ func getRunningPods(dnsName string) (RunningPods, error) {
...
 	if _, exists := runningPods[podName]; !exists {
 		fmt.Printf("Pod (%s) down!  This entry should be removed:  %s:%s\n", podName, account, nodeID)
 		if dryRun == false {
 			controller.UnregisterWithRedis(redisClient, account, nodeID, podName)
+			connCount.unregisterStaleConnectionFromRedis.Inc()
 		}
 	}
 }
 
+func pushValues(cfg *config.Config, connCount *Metrics) {
```
Comment (dehort) [inline cmd/connection_cleaner/main.go:53]: nit-picky:  What do you think about naming this method something a little bit more descriptive?  Something like pushMetricsToPrometheus()?

File: internal/config/config.go [hunk 3]
```diff
@@ -44,6 +44,7 @@ const (
 	GATEWAY_ACTIVE_CONNECTION_REGISTRAR_POLL_MAX_DELAY = "Gateway_Active_Connection_Registrar_Poll_Max_Delay"
 	GATEWAY_CLUSTER_SERVICE_NAME                       = "Gateway_Cluster_Service_Name"
 	NODE_ID                                            = "ReceptorControllerNodeId"
+	PROM_PUSH_GW                                       = "PromPushGW"
```
Comment (dehort) [inline internal/config/config.go:47]: nit-picky:  Can you make this option name a bit more verbose?  Maybe "Prometheus_Push_Gateway"?

File: cmd/connection_cleaner/main.go [hunk 4]
```diff
@@ -33,18 +40,36 @@ func getRunningPods(dnsName string) (RunningPods, error) {
...
+
+}
+
 func main() {
 	var podName = flag.String("pod-name", "receptor-gateway-internal", "DNS name of the internal gateway pod")
 	var dryRun = flag.Bool("dry-run", false, "Just report the stale connections.  No changes will be made.")
+
+	connCount := &Metrics{
+		unregisterStaleConnectionFromRedis: promauto.NewCounter(prometheus.CounterOpts{
+			Name: "receptor_controller_unregister_stale_connection_from_redis_count",
```
Comment (dehort) [inline cmd/connection_cleaner/main.go:68]: Can we change the name to "receptor_controller_connection_cleaner_unregister_stale_connection_from_redis_count"?  I think I'd like to track this count separately from the "usual" method of removing stale connections.

**Review by dehort — CHANGES_REQUESTED** [review]
This looks good overall.  I left a couple of nit-picky comments.  Sorry, I favor being overly verbose in naming things.  I find it helpful when I come back to the code later down the road.  Let me know what you think!

**Comment by dehort** [comment]
Does the push gateway require credentials?

**Comment by SteveHNH** [comment]
> 
> 
> Does the push gateway require credentials?

As far as I know, it does not. We do need to update the network policy to allow our controller to hit it though.

**Pull request merged**